lib/widgets/message_list.dart

```diff
@@ -788,8 +788,16 @@ class _MessageListState extends State<MessageList> with PerAccountStoreAwareStat
 
   @override
   Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
     if (!model.fetched) return const Center(child: CircularProgressIndicator());
 
+    if (model.haveNewest && model.haveOldest
+          && model.messages.isEmpty && model.outboxMessages.isEmpty) {
```

Comment:
Maybe test model.items is empty, rather than messages and outboxMessages separately?
```suggestion
    if (model.items.isEmpty && model.haveNewest && model.haveOldest) {
```

That's the layer the rest of this widget's logic works at: it operates on items and doesn't look at the messages or outboxMessages lists directly.

If in some future we add another way there can be items in the list, with neither messages nor outbox-messages, and don't edit this, then that would mean it'd show the start and end caps, and the items in between. I think that's probably the right outcome: the items are there in the view-model, so should be shown. (If they shouldn't be shown, they shouldn't be in the view-model.)